src/core/meshcop/secure_transport.cpp

```diff
@@ -301,7 +301,8 @@ Error SecureSession::Setup(void)
         mbedtls_ssl_set_timer_cb(&mSsl, this, HandleMbedtlsSetTimer, HandleMbedtlsGetTimer);
     }
 
-#if defined(MBEDTLS_SSL_EXPORT_KEYS) && (MBEDTLS_VERSION_NUMBER >= 0x03000000)
+#if (defined(MBEDTLS_SSL_EXPORT_KEYS) && (MBEDTLS_VERSION_NUMBER >= 0x03000000)) || \
+    (MBEDTLS_VERSION_NUMBER >= 0x03010000)
```

Comment:
This is fine as is, but since we are repeating this combination of checks in a bunch of places, it might help if we define a macro for it and use that instead, something like:

```c++
```c++
#ifdef OT_CONFIG_MBEDTLS_PROVIDES_SSL_KEY_EXPORT
#error "OT_CONFIG_MBEDTLS_PROVIDES_SSL_KEY_EXPORT MUST NOT be defined directly. It is derived from other configs"
#endif

#define OT_CONFIG_MBEDTLS_PROVIDES_SSL_KEY_EXPORT                                   \
    ((defined(MBEDTLS_SSL_EXPORT_KEYS) && (MBEDTLS_VERSION_NUMBER >= 0x03000000)) || \
    (MBEDTLS_VERSION_NUMBER >= 0x03010000))
```

And then use

``` c++
#if OT_CONFIG_MBEDTLS_PROVIDES_SSL_KEY_EXPORT
```

What do you think?

Reply:
We can do it like this:
```c++
#if (defined(MBEDTLS_SSL_EXPORT_KEYS) && (MBEDTLS_VERSION_NUMBER >= 0x03000000)) || \
    (MBEDTLS_VERSION_NUMBER >= 0x03010000)
#define OT_CONFIG_MBEDTLS_PROVIDES_SSL_KEY_EXPORT 1
#else
#define OT_CONFIG_MBEDTLS_PROVIDES_SSL_KEY_EXPORT 0
#endif
```